### PR TITLE
fix bug: job always show success even it failed

### PR DESCRIPTION
**Pull request by Binyang2014**

### CODE REVIEW

**Comment by yqwang-ms** [comment]
Could we also add below stuffs in all runtime bash? So that error will never be ignored again:
```
set -o errexit
set -o nounset
set -o pipefail
```

File: src/kube-runtime/src/runtime [hunk 1]
```diff
@@ -55,12 +55,15 @@ ${PAI_RUNTIME_DIR}/precommands.sh > >(tee -a ${PAI_LOG_DIR}/runtime.pai.stdout)
...
 wait ${USER_PID}
+USER_EXIT_CODE=$?
+echo "[INFO] USER COMMAND END" >> ${PAI_LOG_DIR}/user.pai.stderr
 
 # execute postCommands generated by plugin
-${PAI_RUNTIME_DIR}/postcommands.sh > >(tee -a ${PAI_LOG_DIR}/runtime.pai.stdout) 2> >(tee -a ${PAI_LOG_DIR}/runtime.pai.stderr >&2)
\ No newline at end of file
+${PAI_RUNTIME_DIR}/postcommands.sh > >(tee -a ${PAI_LOG_DIR}/runtime.pai.stdout) 2> >(tee -a ${PAI_LOG_DIR}/runtime.pai.stderr >&2)
+echo "container will exit with code ${USER_EXIT_CODE}" >> ${PAI_LOG_DIR}/runtime.pai.stderr
+exit ${USER_EXIT_CODE}
```
Comment (abuccts) [inline src/kube-runtime/src/runtime]: what if `precommands.sh` or `postcommands.sh` failed? better to add `|| exit $?` at each line's ending

Comment (yqwang-ms) [reply src/kube-runtime/src/runtime]: That likes "set -o errexit", so I suggest add them all in header.

Comment (Binyang2014) [reply src/kube-runtime/src/runtime]: Dian create another PR for this: https://github.com/microsoft/pai/pull/3559, will close this PR

**Pull request closed without merging**